src/ring_theory/ideal/norm.lean

```diff
@@ -0,0 +1,69 @@
+/-
+Copyright (c) 2022 Anne Baanen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen, Alex J. Best
+-/
+
+import linear_algebra.isomorphisms
+import ring_theory.dedekind_domain.ideal
+
+/-!
+# Ideal norms
+This file will define the absolute ideal norm `ideal.abs_norm (I : ideal R) : ℕ` as the cardinality
+of the quotient `R ⧸ I` (setting it to 0 if the cardinality is infinite).
+
+## Main definitions
+ * `submodule.card_quot (S : submodule R M)`: the cardinality of the quotient `M ⧸ S`, in `ℕ`.
+   This maps `⊥` to `0` and `⊤` to `1`.
+
+## TODO
+ * `ideal.abs_norm (I : ideal R)`: the absolute ideal norm, defined as
+   the cardinality of the quotient `R ⧸ I`, as a bundled monoid-with-zero homomorphism.
+   (In an upcoming PR!)
+
+ * Define the relative norm.
+
+-/
+
+open_locale big_operators
+
+namespace submodule
+
+variables {R M : Type*} [ring R] [add_comm_group M] [module R M]
+
+section
+
+open_locale classical
+
+/-- The cardinality of `(M ⧸ S)`, if `(M ⧸ S)` is finite, and `0` otherwise.
+This is used to define the absolute ideal norm `ideal.abs_norm`.
+-/
+noncomputable def card_quot (S : submodule R M) : ℕ :=
+if h : nonempty (fintype (M ⧸ S)) then @fintype.card _ h.some else 0
```

Comment:
Can this be defined in terms of `add_subgroup.index`?

Reply:
Good point, that makes everything a bit simpler!